packages/api/src/controllers/stream.ts

```diff
@@ -233,6 +235,35 @@ async function triggerManyIdleStreamsWebhook(ids: string[], queue: Queue) {
   );
 }
 
+// Reuse the same node for pulling the stream in two cases:
+// 1. if the same pull request is created within 1 min (this prevents race condition between different nodes executing /lockPull)
+// 2. if the same stream was stopped within 10 min (this prevents Mist delays in stopping/starting the stream)
+export function resolvePullUrlFromExistingStreams(
+  existingStreams: DBStream[]
+): { pullUrl: string; pullRegion: string } {
+  if (existingStreams.length !== 1) {
+    return null;
+  }
+  const stream = existingStreams[0];
+  if (
+    stream.pullRegion &&
+    stream.pullLockedBy &&
+    ((stream.pullLockedAt &&
+      stream.pullLockedAt > Date.now() - DEFAULT_PULL_LOCK_LEASE_TIMEOUT) ||
+      (stream.lastSeen &&
+        stream.lastSeen > Date.now() - PULL_REUSE_SAME_NODE_TIMEOUT))
```

Comment:
I wonder if we need to check if the stream already `isActive` as well? Or do we already check this somewhere else? Or do we _not_ want to trust that given the issues we were just discussing, that some of these streams are left for clean-up cause the `/setactive` doesn't work?

Reply:
Yeah, I think that checking the `lastSeen` already covers it. And yeah, we don't tryst `isActive` 🙃